homeassistant/components/mqtt/models.py

```diff
@@ -289,6 +289,30 @@ def async_render_with_possible_json_value(
         return rendered_payload
 
 
+class EntityMonitor:
+    """Monitors entity state changes."""
+
+    _singleton: object = object()
```

Comment:
I think you can use UNDEFINED and UndefinedType here

Reply:
from homeassistant.helpers.typing import UNDEFINED, ConfigType, UndefinedType

Reply:
Is ConfigType not reserved for YAML config

Reply:
Ignore ConfigType, I just grabbed a line from git grep to show where to import it from